### PR TITLE
[Improment](executor)Add downgrade scan thread num by column num

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
refer #35340

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

File: be/src/vec/exec/scan/scanner_context.cpp [hunk 1]
```diff
@@ -80,6 +80,28 @@ ScannerContext::ScannerContext(RuntimeState* state, const TupleDescriptor* outpu
                                                       : state->query_parallel_instance_num());
     _max_thread_num = _max_thread_num == 0 ? 1 : _max_thread_num;
     _max_thread_num = std::min(_max_thread_num, (int32_t)scanners.size());
+
+    // when user not specify scan_thread_num, so we can try downgrade _max_thread_num.
+    // becaue we found in a table with 5k columns, column reader may ocuppy too much memory.
+    // you can refer https://github.com/apache/doris/issues/35340 for details.
+    int32_t max_column_reader_num = state->query_options().max_column_reader_num;
+    if (_state->num_scanner_threads() <= 0 && _max_thread_num != 1 && max_column_reader_num > 0) {
```
Comment (yiguolei) [inline be/src/vec/exec/scan/scanner_context.cpp:88]: do not if here.
Our policy is calculate different number s from user provided and use the minimum one.

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**